### DISCUSSION TITLE
Best practice guide to write new formula

### DISCUSSION THREAD

**Discussion by albertz** [Homebrew/discussions#2574] [Writing Formulae/Casks] [answered]
I want to write a new formula. (For Texmaker specifically, which is based on Qt. But this question should be more generic.)

I guess a good starting point is to use some other similar formula and deviate from that.
I tried to search for `depends_on "qt"` on GitHub but GitHub cannot handle such searches.
So I clicked a bit around and after a while found [qcachegrind](https://github.com/Homebrew/homebrew-core/blob/master/Formula/qcachegrind.rb).

How should I proceed now? I maybe would do this:
```
cd ~/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core

git checkout -B texmaker

edit Formula/texmaker.rb

brew install texmaker  # test that it works...

git add Formula/texmaker.rb
git commit

git push my-fork
```

Is this the way to go? But then after the editing, I would switch back to the master branch?

Is there any documentation on how this would be done?

**Comment by SMillerDev** [comment, marked as answer]
https://docs.brew.sh/Adding-Software-to-Homebrew has the documentation how to add a formula to Homebrew, which is roughly what you did.

**Reply by albertz** [reply]
Hm but when I do `brew install texmaker`, it deletes (stashs away) my changes and switches back to the master branch. So that doesn't work. How do I make it work?

**Reply by SMillerDev** [reply]
Try `brew install --build-from-source texmaker` like in the document

**Comment by albertz** [comment]
Ok, I did the first PR: https://github.com/Homebrew/homebrew-core/pull/90230

**Comment by albertz** [comment]
Note that I had problems with Qt 6 installed via Homebrew. It uses the wrong prefix (`/home/linuxbrew`). I also could not install Qt from source with similar problems. I worked around that for now by creating a `/home/linuxbrew` symlink.

I wanted to post an issue but that doesn't work currently with a 400 error.
